src/meta/src/model/mod.rs

```diff
@@ -51,9 +51,14 @@ pub trait Transactional {
     fn delete_in_transaction(&self, trx: &mut Transaction) -> MetadataModelResult<()>;
 }
 
+/// A marker trait helps to collect all implementors of `MetadataModel` in
+/// `for_all_metadata_models`. The trait should only be implemented by adding item in
+/// `for_all_metadata_models`.
+pub trait MetadataModelMarker {}
```

Comment:
We may define this trait in a private sub-module so that no one can implement it in other places.

<img width="664" alt="image" src="https://user-images.githubusercontent.com/25862682/220560603-c7e9efdd-2a9e-4e69-991e-0c0b8a0ad158.png">

Reply:
Similar to [this](https://github.com/risingwavelabs/risingwave/pull/7752/files/6a328d2c1401db3c1dfbad7c32672f6075b34be1#r1099771724). :)